pkg/kubelet/kubelet_node_status.go

```diff
@@ -328,7 +330,13 @@ func (kl *Kubelet) initialNode(ctx context.Context) (*v1.Node, error) {
 			Effect: v1.TaintEffectNoSchedule,
 		}
 
-		nodeTaints = append(nodeTaints, taint)
+		// Disable functionality in kubelet related to the `--cloud-provider` component flag.
+		if !utilfeature.DefaultFeatureGate.Enabled(features.DisableCloudProviders) {
```

Comment:
I think this is a breaking change -- this feature gate is only for cases where `--cloud-provider=aws|gc|azure|openstack|vsphere`, it shouldn't impact any existing clusters using `--cloud-provider=external`.

Reply:
If there are tests failing when this feature gate is enabled, then the fix is to deploy a CCM that will handle the removal of this taint, or not set `--cloud-provider` at all.

Reply:
@dims @cheftako should we revert this since the CI jobs are still red anyways?

Reply:
@andrewsykim your call. i am fine with which ever way you want to go. Before this change, all tests were failing because the setup failed (nothing could be scheduled). with this change, quite a bit of the tests pass and there is a bucket of tests are red for sure. I did not dig into them.

Reply:
Some tests are expected to pass now because removing the taint in this code path is essentially the same as not setting `--cloud-provider` at all. But this change by itself will be a breaking change for clusters using CCM when this feature gate goes Beta since kubelet's will not a longer set the taint on startup

Reply:
https://github.com/kubernetes/kubernetes/pull/112833